bin/main

```diff
@@ -1,73 +1,73 @@
-end
+board_array = %w[1 2 3 4 5 6 7 8 9]
+board = Board.new(board_array)
 
-def input_request(num)
-  if num.even?
-    puts "it's #{@name_pone} turn"
-    display_board($board)
-    board_input = gets.chomp
-  else
-    puts "it's #{@name_ptwo} turn"
-    display_board($board)
-    board_input = gets.chomp
-  end
-end
+puts 'Welcome to Tic Tac Toe'
+puts "what's player one's name ?"
 
-def welcome
-  puts "Welcome to Tic Tac Toe"
-  puts "what's player one's name ?"
-  @name_pone = gets.chomp
-  puts "what's player two's name ?"
-  @name_ptwo = gets.chomp
-  system('cls') || system('clear')
-end
+player1_name = gets.chomp
+puts "what's player two's name ?"
+
+player2_name = gets.chomp
+players = Players.new(player1_name, player2_name)
+
+class Interface
+  def display_board(board)
+    puts "#{board[0]} | #{board[1]} | #{board[2]}"
+    puts '---------'
+    puts "#{board[3]} | #{board[4]} | #{board[5]}"
+    puts '---------'
+    puts "#{board[6]} | #{board[7]} | #{board[8]} \n"
+  end
 
-def swap_on_board(i, given_number)
-  while $board.any?(given_number)
-    if i.even?
-      $board[given_number.to_i - 1] = 'X'
-    elsif i.odd?
-      $board[given_number.to_i - 1] = 'O'
+  def input_request(num, player1_name, player2_name, board)
+    if num.even?
+      puts "it's #{player1_name} turn"
+    else
+      puts "it's #{player2_name} turn"
     end
+    display_board(board)
+    board_input = gets.chomp
+    board_input
   end
-  $board
 end
 
-welcome
+interface = Interface.new
+
+itr = 0
 
-while i < 9
-  given_number = input_request(i)
-  if given_number.to_i < 10 && $board.any?(given_number.to_s)
-    swap_on_board(i, given_number)
-  	puts "good job \n\n"
-    if given_number.to_i == 7
-      puts "you have won!"
-      break
-    elsif given_number.to_i == 8
-      puts "it's a draw"
-      break
+while itr < 9
+  given_number = interface.input_request(itr, players.player1, players.player2, board.board)
+  if given_number.to_i < 10 && board.board.any?(given_number.to_s)
+    board.swap_on_board(itr, given_number)
+    puts "good job \n\n"
+    if itr == 8
+      system('cls') || system('clear')
+      interface.display_board(board.board)
+      puts "\nIt's a draw there are no more moves"
     end
   else
-    while $board.none?(given_number)
-    	puts "that's not a valid option, try again \n"
-    	#display_board($board)
-      given_number = input_request(i)
+    while board.board.none?(given_number)
+      puts "that's not a valid option, try again \n"
+      given_number = interface.input_request(itr, players.player1, players.player2, board.board)
     end
     puts "Nice move \n\n"
-    swap_on_board(i, given_number)
+    board.swap_on_board(itr, given_number)
+  end
+  if board.winner?(board.board)
+    system('cls') || system('clear')
+    interface.display_board(board.board)
+    if itr.even?
+      puts "The winner is #{players.player1}"
+    else
+      puts "The winner is #{players.payer2}"
```

Comment:
Move the logic into the lib folder please. 
**And for all the other logic, do the same please.**